### DISCUSSION TITLE
HFS as Service on Windows 10

### DISCUSSION THREAD

**Discussion by vcrank74** [rejetto/hfs#692] [Q&A] [answered]
I downloaded version 0.54.0-alpha1. Unpacked it to d:\UserData\Programs\HFS\
Run hfs.exe. Everything works! (inside the network and from outside).

I'm trying to run it as a service https://github.com/rejetto/hfs
`npx qckwinsvc2 install name="HFS" description="HFS" path="%APPDATA%\npm\node_modules\hfs\src\index.js" args="--cwd d:\UserData\Programs\HFS\" now`
It opens via localhost. It doesn't open from a neighboring computer. It doesn't open via the Internet either

**Comment by rejetto** [comment]
Couldn't answer before as I was not home.
Sadly i'm no Windows service expert.
When you access on localhost, see what's in the Admin-panel > logs > console

**Comment by vcrank74** [comment]
There are only old entries from the end of July.

**Comment by rejetto** [comment]
that's not a problem.
the content will tell you what network is being served.

**Comment by vcrank74** [comment]
Run as service:
![as_service](https://github.com/user-attachments/assets/c4853307-d0c2-43b1-b966-d30c22947bf2)
Uninstall service, run hfs.exe:
![standalone](https://github.com/user-attachments/assets/9bb3d7a0-f569-4e94-9794-9d4047d75329)

**Comment by rejetto** [comment]
<img width="332" alt="image" src="https://github.com/user-attachments/assets/4b7db30e-9d37-4422-84f6-2f656bb78442">

apparently, when running as a service, your HFS is not only unable to receive connections (inbound), but neither can make outbound connections.
This is clearly a system problem, i mean, the system is not allowing network activity to the service.

Try modifying the HFS service to use your personal account instead instead of "local system", in this window
<img width="484" alt="image" src="https://github.com/user-attachments/assets/c231baec-3d20-4f08-900a-e50c180c3ec9">

**Comment by vcrank74** [comment]
I did. But it didn't help. :(
![as_service_local_user](https://github.com/user-attachments/assets/f6d62fa3-98d6-41f7-b046-3dce76d3931a)

**Reply by STaylor135** [reply]
Have you made an explicit allowance for the program in the software firewall? 

You can probably find instructions online but I'll run this down quickly.  I'm going to assume you're running windows 8 or later, it should all be the same and might work for 7 but I don't remember and that the windows firewall is on. 

Open the start menu or right click the start menu button and select run
type: 
firewall.cpl
Open that file. 
On the left you'll see the option: "Allow an app or feature through Windows Defender Firewall"
Click that link
On the new screen click Change settings
On the lower part of the window select Allow another app
In the new window click the browse button and navigate to and select the location of hfs.exe
*****
In looking at your post I see you're running a JS directly, I've not done this and honestly I'd recommend running the exe in windows in this situation. If you do want to run the JS, you may need to allow the interpreter these permissions, I might be able to help with that too
*****
Once you select it, before you close the "Add an app" screen, select the network types button and make sure both private and public are selected.
Click okay
Click add

Now the program should be allowed access both out and in
I don't think you'll need to make a specific rule for the ports you use.

**Reply by cbr00t** [reply, marked as answer]
You may also allow access to "node.exe" application from windows firewall in addition to hfs.exe.
'npx' runs a "node.exe" process.
("c:\Program Files\nodejs\node.exe")
